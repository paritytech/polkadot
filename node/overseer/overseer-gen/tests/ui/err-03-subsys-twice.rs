#![allow(dead_code)]

use polkadot_overseer_gen::*;

#[derive(Default)]
struct AwesomeSubSys;

#[derive(Clone, Debug)]
struct SigSigSig;

struct Event;

#[derive(Clone, Debug)]
struct MsgStrukt(u8);

#[derive(Clone, Debug)]
struct MsgStrukt2(f64);

#[overlord(signal=SigSigSig, event=Event, gen=AllMessages, error=OverseerError)]
struct Overseer {
	#[subsystem(MsgStrukt)]
	sub0: AwesomeSubSys,

	#[subsystem(MsgStrukt2)]
	sub1: AwesomeSubSys,
}

#[derive(Debug, Clone)]
struct DummySpawner;

struct DummyCtx;

fn main() {
	let overseer = Overseer::<_,_>::builder()
		.sub0(AwesomeSubSys::default())
		.i_like_pie(std::f64::consts::PI)
		.spawner(DummySpawner)
		.build(|| -> DummyCtx { DummyCtx } );
}
