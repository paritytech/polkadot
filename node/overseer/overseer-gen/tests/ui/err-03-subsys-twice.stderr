error: Duplicate subsystem names `AwesomeSubSys`
  --> $DIR/err-03-subsys-twice.rs:25:8
   |
25 |     sub1: AwesomeSubSys,
   |           ^^^^^^^^^^^^^

error: previously defined here.
  --> $DIR/err-03-subsys-twice.rs:22:8
   |
22 |     sub0: AwesomeSubSys,
   |           ^^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared type `Overseer`
  --> $DIR/err-03-subsys-twice.rs:34:17
   |
34 |     let overseer = Overseer::<_,_>::builder()
   |                    ^^^^^^^^ use of undeclared type `Overseer`
